ts/packages/anchor/src/coder/borsh/instruction.ts

```diff
@@ -303,11 +304,15 @@ class InstructionFormatter {
           const variants = typeDef.type.variants;
           const variant = Object.keys(data)[0];
           const enumType = data[variant];
+          const camelCaseVariant = camelCase(variant);
+          const enumVariant = variants.find((v) => camelCase(v.name) === camelCaseVariant);
+          const relevantFields = enumVariant?.fields;
```

Comment:
Saw we were converting to camelCases at many places. There was a case for example, where Swap enum name `Openbook` on [this contract](https://explorer.solana.com/address/JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4/anchor-program) was re-cased to `openbook` post instruction parsing, and missed matching as a result

Reply:
Thanks for the example, I wouldn't be surprised by this since case conversion has always been problematic.